webpack/assets/javascripts/react_app/components/ColumnSelector/ColumnSelector.js

```diff
@@ -32,8 +33,10 @@ const ColumnSelector = props => {
   };
 
   async function updateTablePreference() {
+    if (!url || !controller) return;
+    setSaving(true);
     if (!hasPreference) {
-      await API.post(url, { name: 'hosts', columns: getColumnKeys() });
+      await API.post(url, { name: controller, columns: getColumnKeys() });
```

Comment:
I'm not sure why "hosts" was hard-coded here, but this should make this component completely reusable.

Reply:
This uses `API.post` directly, which is way outside the normal Foreman pattern of using Redux, but I wanted to get this functioning and just reuse the component without too many changes.